### PR TITLE
Adding assert statement to spatial dropout and virtual batch norm

### PR DESCRIPTION
**Pull request by NippunSharma**
I have added an assert statement in `spatial_dropout_impl.hpp` and `virtual_batch_norm_impl.hpp` that asserts that the input size must be divisible by the number of input maps. Like there is one in `batch_norm_impl.hpp`.

### CODE REVIEW

**Review by zoq — APPROVED** [review]
Looks good to me, thanks for putting this together.

**Review by kartikdutt18 — APPROVED** [review]
LGTM! Thanks for working on this.
